compiler/control/OMROptions.cpp

```diff
@@ -2217,6 +2218,14 @@ OMR::Options::jitLatePostProcess(TR::OptionSet *optionSet, void * jitConfig)
             if (_coldUpgradeSampleThreshold == TR_DEFAULT_COLD_UPGRADE_SAMPLE_THRESHOLD)
                _coldUpgradeSampleThreshold = 2;
             }
+         else
+            {
+            // For AOT with GCR enabled we can be more conservative with 
+            // upgrades through sampling because we can rely on GCR
```

Comment:
The new option is going to be checked in the downstream project, but regardless of the new option's value, if AOT and GCR are enabled, upgrading through sampling is going to be more conservative. I felt this is a good change to make in any case.
As I mentioned in private conversation, upgrades during startup can cause temporary backups of compilations: once the upgrade starts the JIT is not going to cancel it and while this upgrade is in progress many AOT loads are just waiting to be server (GCR recomps don't have this problem because they are allowed only after startup).
With this solution we allow upgrades through sampling only for methods that receive a "significant" amount of samples and thus they should be worthy of an upgrade.